Fix mine layer desync.

pchote opened the pull request:
Fixes #17899.

Mailaender commented:
Is https://github.com/OpenRA/OpenRA/blob/6ba02800ab5dfd07a1f50fd35b391604b8e40255/OpenRA.Mods.Cnc/Traits/Minelayer.cs#L118 then also wrong?

pchote committed:
Fix mine layer desync.

World.FogObscures depends on the local RenderPlayer and should not
be used from simulation code!

pchote commented:
Yes. Fixed.

abcdefg30 commented:
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/33eeb423c054244a9b51b83b39bc543347036dd0...435481db91474390e2a485d15a57fc0e93dfb67)

https://github.com/OpenRA/OpenRA/commit/798c92abeb154589c86eb339dbaef47c2e7abb23
